# HG changeset patch
# User HackaLittleBit
# Date 1388177822 0
#      Fri Dec 27 20:57:02 2013 +0000
# Node ID 7b8910ca19b87e7af115c3c337f30a8f9fcb8312
# Parent  56e89ad25068bb661c12da332b1c2d08d013c97d
Reinstate holding speed for aircrafts + improved holding patterns.

diff --git a/src/table/airport_movement.h b/src/table/airport_movement.h
--- a/src/table/airport_movement.h
+++ b/src/table/airport_movement.h
@@ -45,7 +45,7 @@
 };
 
 /** Country Airfield (small) 4x3. */
-static const AirportMovingData _airport_moving_data_country[22] = {
+static const AirportMovingData _airport_moving_data_country[26] = {
 	AMD(   53,    3, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar
 	AMD(   53,   27, 0,                               DIR_N ), // 01 Taxi to right outside depot
 	AMD(   32,   23, AMED_EXACTPOS,                   DIR_NW), // 02 Terminal 1
@@ -56,22 +56,26 @@
 	AMD(   61,   40, AMED_EXACTPOS,                   DIR_NE), // 07 Taxi to start of runway (takeoff)
 	AMD(    3,   40, AMED_NOSPDCLAMP,                 DIR_N ), // 08 Accelerate to end of runway
 	AMD(  -79,   40, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 09 Take off
-	AMD(  177,   40, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 10 Fly to landing position in air
-	AMD(   56,   40, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 11 Going down for land
+	AMD(  177,   40, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 10 Fly to landing position in air
+	AMD(   56,   40, AMED_HOLD       | AMED_LAND,     DIR_N ), // 11 Going down for land
 	AMD(    3,   40, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 12 Just landed, brake until end of runway
 	AMD(    7,   40, 0,                               DIR_N ), // 13 Just landed, turn around and taxi 1 square
 	AMD(   53,   40, 0,                               DIR_N ), // 14 Taxi from runway to crossing
-	AMD(    1,  193, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 15 Fly around waiting for a landing spot (north-east)
-	AMD(    1,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 16 Fly around waiting for a landing spot (north-west)
-	AMD(  257,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 17 Fly around waiting for a landing spot (south-west)
-	AMD(  273,   47, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 18 Fly around waiting for a landing spot (south)
+	AMD(    0,  150, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 15 Fly around waiting for a landing spot (north-east)
+	AMD(    0,  -17, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 16 Fly around waiting for a landing spot (north-west)
+	AMD(  232,  -17, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 17 Fly around waiting for a landing spot (south-west)
+	AMD(  232,   40, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 18 Fly around waiting for a landing spot (south)
 	AMD(   44,   37, AMED_HELI_RAISE,                 DIR_N ), // 19 Helicopter takeoff
 	AMD(   44,   40, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 20 In position above landing spot helicopter
 	AMD(   44,   40, AMED_HELI_LOWER,                 DIR_N ), // 21 Helicopter landing
+	AMD(  365,   40, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 22 Initial DIAGDIR_NE approach
+	AMD(   56, -193, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 23 Initial DIAGDIR_SE approach
+	AMD(  -17,  -17, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 24 Initial DIAGDIR_SW approach
+	AMD(	0,  235, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 25 Initial DIAGDIR_NW approach
 };
 
 /** Commuter Airfield (small) 5x4. */
-static const AirportMovingData _airport_moving_data_commuter[37] = {
+static const AirportMovingData _airport_moving_data_commuter[41] = {
 	AMD(   69,    3, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar
 	AMD(   72,   22, 0,                               DIR_N ), // 01 Taxi to right outside depot
 	AMD(    8,   22, AMED_EXACTPOS,                   DIR_SW), // 01 Taxi to right outside depot
@@ -88,15 +92,15 @@
 	AMD(    7,   54, AMED_NOSPDCLAMP,                 DIR_N ), // 13 Release control of runway, for smoother movement
 	AMD(    5,   54, AMED_NOSPDCLAMP,                 DIR_N ), // 14 End of runway
 	AMD(  -79,   54, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 15 Take off
-	AMD(  145,   54, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 16 Fly to landing position in air
-	AMD(   73,   54, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 17 Going down for land
+	AMD(  178,   54, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 16 Fly to landing position in air
+	AMD(   73,   54, AMED_HOLD       | AMED_LAND,     DIR_N ), // 17 Going down for land
 	AMD(    3,   54, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 18 Just landed, brake until end of runway
 	AMD(   12,   54, AMED_SLOWTURN,                   DIR_NW), // 19 Just landed, turn around and taxi
 	AMD(    8,   32, 0,                               DIR_NW), // 20 Taxi from runway to crossing
-	AMD(    1,  149, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (north-east)
-	AMD(    1,    6, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 22 Fly around waiting for a landing spot (north-west)
-	AMD(  193,    6, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 23 Fly around waiting for a landing spot (south-west)
-	AMD(  225,   62, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 24 Fly around waiting for a landing spot (south)
+	AMD(    0,  149, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (north-east)
+	AMD(    0,  -14, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 22 Fly around waiting for a landing spot (north-west)
+	AMD(  225,  -14, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 23 Fly around waiting for a landing spot (south-west)
+	AMD(  225,   54, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 24 Fly around waiting for a landing spot (south)
 	/* Helicopter */
 	AMD(   80,    0, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 25 Bufferspace before helipad
 	AMD(   80,    0, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 26 Bufferspace before helipad
@@ -108,12 +112,16 @@
 	AMD(   48,    8, AMED_HELI_RAISE,                 DIR_N ), // 32 Takeoff Helipad2
 	AMD(   64,   22, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 33 Go to position for Hangarentrance in air
 	AMD(   64,   22, AMED_HELI_LOWER,                 DIR_N ), // 34 Land in front of hangar
-	AMD(   40,    8, AMED_EXACTPOS,                   DIR_N ), // pre-helitakeoff helipad 1
-	AMD(   56,    8, AMED_EXACTPOS,                   DIR_N ), // pre-helitakeoff helipad 2
+	AMD(   40,    8, AMED_EXACTPOS,                   DIR_N ), // 35 pre-helitakeoff helipad 1
+	AMD(   56,    8, AMED_EXACTPOS,                   DIR_N ), // 36 pre-helitakeoff helipad 2
+	AMD(  365,   54, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 37 Initial DIAGDIR_NE approach
+	AMD(   56, -193, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 38 Initial DIAGDIR_SE approach
+	AMD(  -19,  -14, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 39 Initial DIAGDIR_SW approach
+	AMD(	0,  240, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 40 Initial DIAGDIR_NW approach
 };
 
 /** City Airport (large) 6x6. */
-static const AirportMovingData _airport_moving_data_city[] = {
+static const AirportMovingData _airport_moving_data_city[30] = {
 	AMD(   85,    3, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar
 	AMD(   85,   22, 0,                               DIR_N ), // 01 Taxi to right outside depot
 	AMD(   26,   41, AMED_EXACTPOS,                   DIR_SW), // 02 Terminal 1
@@ -132,22 +140,22 @@
 	AMD(   20,   87, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 15 Just landed, brake until end of runway
 	AMD(   20,   87, 0,                               DIR_N ), // 16 Just landed, turn around and taxi 1 square // NOT USED
 	AMD(   36,   71, 0,                               DIR_N ), // 17 Taxi from runway to crossing
-	AMD(  160,   87, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 18 Fly around waiting for a landing spot (north-east)
-	AMD(  140,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 19 Final approach fix
-	AMD(  257,    1, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 20 Fly around waiting for a landing spot (south-west)
-	AMD(  273,   49, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (south)
+	AMD(  144,   87, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 18 Fly around waiting for a landing spot (north-east)
+	AMD(  140, -200, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 19 Not Used
+	AMD(  257,    0, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 20 Fly around waiting for a landing spot (south-west)
+	AMD(  273,   68, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (south)
 	AMD(   44,   63, AMED_HELI_RAISE,                 DIR_N ), // 22 Helicopter takeoff
 	AMD(   28,   74, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 23 In position above landing spot helicopter
 	AMD(   28,   74, AMED_HELI_LOWER,                 DIR_N ), // 24 Helicopter landing
-	AMD(  145,    1, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 25 Fly around waiting for a landing spot (north-west)
-	AMD(  -32,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 26 Initial approach fix (north)
-	AMD(  300,  -48, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 27 Initial approach fix (south)
-	AMD(  140,  -48, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 28 Intermediate Approach fix (south), IAF (west)
-	AMD(  -32,  120, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 29 Initial approach fix (east)
+	AMD(  144,    0, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 25 Fly around waiting for a landing spot (north-west)
+	AMD(  396,  157, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 26 Initial DIAGDIR_NE approach
+	AMD(  144, -180, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 27 Initial DIAGDIR_SE approach
+	AMD(    0,    0, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 28 Initial DIAGDIR_SW approach
+	AMD(	0,  172, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 29 Initial DIAGDIR_NW approach
 };
 
 /** Metropolitan Airport (metropolitan) - 2 runways. */
-static const AirportMovingData _airport_moving_data_metropolitan[28] = {
+static const AirportMovingData _airport_moving_data_metropolitan[32] = {
 	AMD(   85,    3, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar
 	AMD(   85,   22, 0,                               DIR_N ), // 01 Taxi to right outside depot
 	AMD(   26,   41, AMED_EXACTPOS,                   DIR_SW), // 02 Terminal 1
@@ -161,25 +169,29 @@
 	AMD(   89,   69, AMED_EXACTPOS,                   DIR_NE), // 10 Taxi to start of runway (takeoff)
 	AMD(    3,   69, AMED_NOSPDCLAMP,                 DIR_N ), // 11 Accelerate to end of runway
 	AMD(  -79,   69, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 12 Take off
-	AMD(  177,   85, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 13 Fly to landing position in air
-	AMD(   89,   85, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 14 Going down for land
+	AMD(  205,   85, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 13 Fly to landing position in air
+	AMD(   89,   85, AMED_HOLD       | AMED_LAND,     DIR_N ), // 14 Going down for land
 	AMD(    3,   85, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 15 Just landed, brake until end of runway
 	AMD(   21,   85, 0,                               DIR_N ), // 16 Just landed, turn around and taxi 1 square
 	AMD(   21,   69, 0,                               DIR_N ), // 17 On Runway-out taxiing to In-Way
 	AMD(   21,   58, AMED_EXACTPOS,                   DIR_SW), // 18 Taxi from runway to crossing
-	AMD(    1,  193, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 19 Fly around waiting for a landing spot (north-east)
-	AMD(    1,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 20 Fly around waiting for a landing spot (north-west)
-	AMD(  257,    1, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (south-west)
-	AMD(  273,   49, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 22 Fly around waiting for a landing spot (south)
+	AMD(    1,  193, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 19 Fly around waiting for a landing spot (north-east)
+	AMD(    1,    1, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 20 Fly around waiting for a landing spot (north-west)
+	AMD(  257,    1, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 21 Fly around waiting for a landing spot (south-west)
+	AMD(  273,   49, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 22 Fly around waiting for a landing spot (south)
 	AMD(   44,   58, 0,                               DIR_N ), // 23 Helicopter takeoff spot on ground (to clear airport sooner)
 	AMD(   44,   63, AMED_HELI_RAISE,                 DIR_N ), // 24 Helicopter takeoff
 	AMD(   15,   54, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 25 Get in position above landing spot helicopter
 	AMD(   15,   54, AMED_HELI_LOWER,                 DIR_N ), // 26 Helicopter landing
 	AMD(   21,   58, AMED_EXACTPOS,                   DIR_SW), // 27 Transitions after landing to on-ground movement
+	AMD(  396,  157, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 28 Initial DIAGDIR_NE approach
+	AMD(  144, -180, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 29 Initial DIAGDIR_SE approach
+	AMD(    0,  -64, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 30 Initial DIAGDIR_SW approach
+	AMD(	0,  172, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 31 Initial DIAGDIR_NW approach
 };
 
 /** International Airport (international) - 2 runways, 6 terminals, dedicated helipad. */
-static const AirportMovingData _airport_moving_data_international[51] = {
+static const AirportMovingData _airport_moving_data_international[55] = {
 	AMD(    7,   55, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar 1
 	AMD(  100,   21, AMED_EXACTPOS,                   DIR_SE), // 01 In Hangar 2
 	AMD(    7,   70, 0,                               DIR_N ), // 02 Taxi to right outside depot
@@ -212,15 +224,15 @@
 	AMD(   60,    6, AMED_NOSPDCLAMP,                 DIR_N ), // 29 Release control of runway, for smoother movement
 	AMD(  105,    6, AMED_NOSPDCLAMP,                 DIR_N ), // 30 End of runway
 	AMD(  190,    6, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 31 Take off
-	AMD(  193,  104, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 32 Fly to landing position in air
-	AMD(  105,  104, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 33 Going down for land
+	AMD(  193,  104, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 32 Fly to landing position in air
+	AMD(  105,  104, AMED_HOLD       | AMED_LAND,     DIR_N ), // 33 Going down for land
 	AMD(    3,  104, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 34 Just landed, brake until end of runway
 	AMD(   12,  104, AMED_SLOWTURN,                   DIR_N ), // 35 Just landed, turn around and taxi 1 square
 	AMD(    7,   84, 0,                               DIR_N ), // 36 Taxi from runway to crossing
-	AMD(    1,  209, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 37 Fly around waiting for a landing spot (north-east)
-	AMD(    1,    6, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 38 Fly around waiting for a landing spot (north-west)
-	AMD(  273,    6, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 39 Fly around waiting for a landing spot (south-west)
-	AMD(  305,   81, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 40 Fly around waiting for a landing spot (south)
+	AMD(    1,  209, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 37 Fly around waiting for a landing spot (north-east)
+	AMD(    1,    6, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 38 Fly around waiting for a landing spot (north-west)
+	AMD(  273,    6, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 39 Fly around waiting for a landing spot (south-west)
+	AMD(  305,   81, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 40 Fly around waiting for a landing spot (south)
 	/* Helicopter */
 	AMD(  128,   80, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 41 Bufferspace before helipad
 	AMD(  128,   80, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 42 Bufferspace before helipad
@@ -232,10 +244,14 @@
 	AMD(  104,   55, AMED_HELI_RAISE,                 DIR_N ), // 48 Takeoff Helipad2
 	AMD(  104,   32, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 49 Go to position for Hangarentrance in air
 	AMD(  104,   32, AMED_HELI_LOWER,                 DIR_N ), // 50 Land in HANGAR2_AREA to go to hangar
+	AMD(  396,  157, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 51 Initial DIAGDIR_NE approach
+	AMD(  155, -126, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 52 Initial DIAGDIR_SE approach
+	AMD(  124,  -64, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 53 Initial DIAGDIR_SW approach
+	AMD(  -44,  195, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 54 Initial DIAGDIR_NW approach
 };
 
 /** Intercontinental Airport - 4 runways, 8 terminals, 2 dedicated helipads. */
-static const AirportMovingData _airport_moving_data_intercontinental[77] = {
+static const AirportMovingData _airport_moving_data_intercontinental[81] = {
 	AMD(    8,   87, AMED_EXACTPOS,                   DIR_SE), // 00 In Hangar 1
 	AMD(  136,   72, AMED_EXACTPOS,                   DIR_SE), // 01 In Hangar 2
 	AMD(    8,  104, 0,                               DIR_N ), // 02 Taxi to right outside depot 1
@@ -272,17 +288,17 @@
 	AMD(  119,   24, AMED_NOSPDCLAMP,                 DIR_N ), // 33 Release control of runway, for smoother movement
 	AMD(  117,   24, AMED_NOSPDCLAMP,                 DIR_N ), // 34 End of runway
 	AMD(  197,   24, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 35 Take off
-	AMD(  254,   84, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 36 Flying to landing position in air
-	AMD(  117,  168, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 37 Going down for land
+	AMD(  254,   84, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 36 Flying to landing position in air
+	AMD(  117,  168, AMED_HOLD       | AMED_LAND,     DIR_N ), // 37 Going down for land
 	AMD(    8,  168, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 38 Just landed, brake until end of runway
 	AMD(    8,  168, 0,                               DIR_N ), // 39 Just landed, turn around and taxi
 	AMD(    8,  144, 0,                               DIR_NW), // 40 Taxi from runway
 	AMD(    8,  128, 0,                               DIR_NW), // 41 Taxi from runway
 	AMD(    8,  120, AMED_EXACTPOS,                   DIR_NW), // 42 Airport entrance
-	AMD(   56,  344, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 43 Fly around waiting for a landing spot (north-east)
-	AMD( -200,   88, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 44 Fly around waiting for a landing spot (north-west)
-	AMD(   56, -168, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 45 Fly around waiting for a landing spot (south-west)
-	AMD(  312,   88, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 46 Fly around waiting for a landing spot (south)
+	AMD(   56,  344, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 43 Fly around waiting for a landing spot (north-east)
+	AMD( -200,   88, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 44 Fly around waiting for a landing spot (north-west)
+	AMD(   56, -168, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 45 Fly around waiting for a landing spot (south-west)
+	AMD(  312,   88, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 46 Fly around waiting for a landing spot (south)
 	/* Helicopter */
 	AMD(   96,   40, AMED_NOSPDCLAMP,                 DIR_N ), // 47 Bufferspace before helipad
 	AMD(   96,   40, AMED_NOSPDCLAMP,                 DIR_N ), // 48 Bufferspace before helipad
@@ -300,20 +316,24 @@
 	AMD(   16,  152, AMED_NOSPDCLAMP,                 DIR_N ), // 60 Release control of runway2, for smoother movement
 	AMD(   20,  152, AMED_NOSPDCLAMP,                 DIR_N ), // 61 End of runway2
 	AMD(  -56,  152, AMED_NOSPDCLAMP | AMED_TAKEOFF,  DIR_N ), // 62 Take off2
-	AMD(   24,    8, AMED_NOSPDCLAMP | AMED_LAND,     DIR_N ), // 63 Going down for land2
+	AMD(   24,    8, AMED_HOLD       | AMED_LAND,     DIR_N ), // 63 Going down for land2
 	AMD(  136,    8, AMED_NOSPDCLAMP | AMED_BRAKE,    DIR_N ), // 64 Just landed, brake until end of runway2in
 	AMD(  136,    8, 0,                               DIR_N ), // 65 Just landed, turn around and taxi
 	AMD(  136,   24, 0,                               DIR_SE), // 66 Taxi from runway 2in
 	AMD(  136,   40, 0,                               DIR_SE), // 67 Taxi from runway 2in
 	AMD(  136,   56, AMED_EXACTPOS,                   DIR_NE), // 68 Airport entrance2
-	AMD(  -56,    8, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 69 Fly to landing position in air2
+	AMD(  -56,    8, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 69 Fly to landing position in air2
 	AMD(   88,   40, 0,                               DIR_N ), // 70 Taxi Term group 2 exit - opp heli1
 	AMD(   72,   40, 0,                               DIR_N ), // 71 Taxi Term group 2 exit - opp heli2
 	AMD(   88,   57, AMED_EXACTPOS,                   DIR_SE), // 72 pre-helitakeoff helipad 1
 	AMD(   71,   56, AMED_EXACTPOS,                   DIR_NE), // 73 pre-helitakeoff helipad 2
 	AMD(    8,  120, AMED_HELI_RAISE,                 DIR_N ), // 74 Helitakeoff outside depot 1
 	AMD(  136,  104, AMED_HELI_RAISE,                 DIR_N ), // 75 Helitakeoff outside depot 2
-	AMD(  197,  168, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 76 Fly to landing position in air1
+	AMD(  197,  168, AMED_HOLD       | AMED_SLOWTURN, DIR_N ), // 76 Fly to landing position in air1
+	AMD(  393,   88, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 77 Initial DIAGDIR_NE approach
+	AMD(  312,    8, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 78 Initial DIAGDIR_SE approach
+	AMD( -278,   88, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 79 Initial DIAGDIR_SW approach
+	AMD( -200,  162, AMED_NOSPDCLAMP | AMED_SLOWTURN, DIR_N ), // 80 Initial DIAGDIR_NW approach
 };
 
 
@@ -419,7 +439,7 @@
  * this may be changed later when airports are moved to external file  */
 static const HangarTileTable _airport_depots_country[] = { {{3, 0}, DIR_SE, 0} };
 static const byte _airport_terminal_country[] = {1, 2};
-static const byte _airport_entries_country[] = {16, 15, 18, 17};
+static const byte _airport_entries_country[] = {22, 23, 24, 25};
 static const AirportFTAbuildup _airport_fta_country[] = {
 	{  0, HANGAR, NOTHING_block, 1 },
 	{  1, 255, AIRPORT_BUSY_block, 0 }, { 1, HANGAR, 0, 0 }, { 1, TERM1, TERM1_block, 2 }, { 1, TERM2, 0, 4 }, { 1, HELITAKEOFF, 0, 19 }, { 1, 0, 0, 6 },
@@ -446,12 +466,16 @@
 	{ 19, HELITAKEOFF, NOTHING_block, 0 },
 	{ 20, HELILANDING, AIRPORT_BUSY_block, 21 },
 	{ 21, HELIENDLANDING, AIRPORT_BUSY_block, 1 },
+	{ 22, 0, NOTHING_block, 10},
+	{ 23, 0, NOTHING_block, 17},
+	{ 24, 0, NOTHING_block, 17},
+	{ 25, 0, NOTHING_block, 16},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
 static const HangarTileTable _airport_depots_commuter[] = { {{4, 0}, DIR_SE, 0} };
 static const byte _airport_terminal_commuter[] = { 1, 3 };
-static const byte _airport_entries_commuter[] = {22, 21, 24, 23};
+static const byte _airport_entries_commuter[] = {37, 38, 39, 40};
 static const AirportFTAbuildup _airport_fta_commuter[] = {
 	{  0, HANGAR, NOTHING_block, 1 }, { 0, HELITAKEOFF, HELIPAD2_block, 1 }, { 0, 0, 0, 1 },
 	{  1, 255, TAXIWAY_BUSY_block, 0 }, { 1, HANGAR, 0, 0 }, { 1, DEPART, 0, 11 }, { 1, TERM1, TAXIWAY_BUSY_block, 10 }, { 1, TERM2, TAXIWAY_BUSY_block, 10 }, { 1, TERM3, TAXIWAY_BUSY_block, 10 }, { 1, HELIPAD1, TAXIWAY_BUSY_block, 10 }, { 1, HELIPAD2, TAXIWAY_BUSY_block, 10 }, { 1, HELITAKEOFF, TAXIWAY_BUSY_block, 10 }, { 1, 0, 0, 0 },
@@ -496,12 +520,16 @@
 	{ 34, 0, TAXIWAY_BUSY_block, 1 },
 	{ 35, 0, HELIPAD1_block, 31 },
 	{ 36, 0, HELIPAD2_block, 32 },
+	{ 37, 0, NOTHING_block, 16},
+	{ 38, 0, NOTHING_block, 23},
+	{ 39, 0, NOTHING_block, 23},
+	{ 40, 0, NOTHING_block, 22},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
 static const HangarTileTable _airport_depots_city[] = { {{5, 0}, DIR_SE, 0} };
 static const byte _airport_terminal_city[] = { 1, 3 };
-static const byte _airport_entries_city[] = {26, 29, 27, 28};
+static const byte _airport_entries_city[] = {26, 27, 28, 29};
 static const AirportFTAbuildup _airport_fta_city[] = {
 	{  0, HANGAR, NOTHING_block, 1 }, { 0, DEPART, OUT_WAY_block, 1 }, { 0, 0, 0, 1 },
 	{  1, 255, TAXIWAY_BUSY_block, 0 }, { 1, HANGAR, 0, 0 }, { 1, TERM2, 0, 6 }, { 1, TERM3, 0, 6 }, { 1, 0, 0, 7 }, // for all else, go to 7
@@ -533,16 +561,16 @@
 	{ 23, HELILANDING, IN_WAY_block, 24 },
 	{ 24, HELIENDLANDING, IN_WAY_block, 17 },
 	{ 25, 0, NOTHING_block, 20},
-	{ 26, 0, NOTHING_block, 19},
-	{ 27, 0, NOTHING_block, 28},
-	{ 28, 0, NOTHING_block, 19},
-	{ 29, 0, NOTHING_block, 26},
+	{ 26, 0, NOTHING_block, 13},
+	{ 27, 0, NOTHING_block, 20},
+	{ 28, 0, NOTHING_block, 20},
+	{ 29, 0, NOTHING_block, 28},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
 static const HangarTileTable _airport_depots_metropolitan[] = { {{5, 0}, DIR_SE, 0} };
 static const byte _airport_terminal_metropolitan[] = { 1, 3 };
-static const byte _airport_entries_metropolitan[] = {20, 19, 22, 21};
+static const byte _airport_entries_metropolitan[] = {28, 29, 30, 31};
 static const AirportFTAbuildup _airport_fta_metropolitan[] = {
 	{  0, HANGAR, NOTHING_block, 1 },
 	{  1, 255, TAXIWAY_BUSY_block, 0 }, { 1, HANGAR, 0, 0 }, { 1, TERM2, 0, 6 }, { 1, TERM3, 0, 6 }, { 1, 0, 0, 7 }, // for all else, go to 7
@@ -576,12 +604,16 @@
 	{ 25, HELILANDING, IN_WAY_block, 26 },
 	{ 26, HELIENDLANDING, IN_WAY_block, 18 },
 	{ 27, 255, TAXIWAY_BUSY_block, 27 }, { 27, TERM1, TERM1_block, 2 }, { 27, 0, 0, 7 },
+	{ 28, 0, NOTHING_block, 13},
+	{ 29, 0, NOTHING_block, 21},
+	{ 30, 0, NOTHING_block, 21},
+	{ 31, 0, NOTHING_block, 20},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
 static const HangarTileTable _airport_depots_international[] = { {{0, 3}, DIR_SE, 0}, {{6, 1}, DIR_SE, 1} };
 static const byte _airport_terminal_international[] = { 2, 3, 3 };
-static const byte _airport_entries_international[] = { 38, 37, 40, 39 };
+static const byte _airport_entries_international[] = { 51, 52, 53, 54 };
 static const AirportFTAbuildup _airport_fta_international[] = {
 	{  0, HANGAR, NOTHING_block, 2 }, { 0, 255, TERM_GROUP1_block, 0 }, { 0, 255, TERM_GROUP2_ENTER1_block, 1 }, { 0, HELITAKEOFF, HELIPAD1_block, 2 }, { 0, 0, 0, 2 },
 	{  1, HANGAR, NOTHING_block, 3 }, { 1, 255, HANGAR2_AREA_block, 1 }, { 1, HELITAKEOFF, HELIPAD2_block, 3 }, { 1, 0, 0, 3 },
@@ -640,13 +672,17 @@
 	{ 48, HELITAKEOFF, NOTHING_block, 0 },
 	{ 49, 0, HANGAR2_AREA_block, 50 }, // need to go to hangar when waiting in air
 	{ 50, 0, HANGAR2_AREA_block, 3 },
+	{ 51, 0, NOTHING_block, 32},
+	{ 52, 0, NOTHING_block, 39},
+	{ 53, 0, NOTHING_block, 39},
+	{ 54, 0, NOTHING_block, 38},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
 /* intercontinental */
 static const HangarTileTable _airport_depots_intercontinental[] = { {{0, 5}, DIR_SE, 0}, {{8, 4}, DIR_SE, 1} };
 static const byte _airport_terminal_intercontinental[] = { 2, 4, 4 };
-static const byte _airport_entries_intercontinental[] = { 44, 43, 46, 45 };
+static const byte _airport_entries_intercontinental[] = { 77, 78, 79, 80 };
 static const AirportFTAbuildup _airport_fta_intercontinental[] = {
 	{  0, HANGAR, NOTHING_block, 2 }, { 0, 255, HANGAR1_AREA_block | TERM_GROUP1_block, 0 }, { 0, 255, HANGAR1_AREA_block | TERM_GROUP1_block, 1 }, { 0, DEPART, HANGAR1_AREA_block | TERM_GROUP1_block, 2 }, { 0, 0, 0, 2 },
 	{  1, HANGAR, NOTHING_block, 3 }, { 1, 255, HANGAR2_AREA_block, 1 }, { 1, 255, HANGAR2_AREA_block, 0 }, { 1, 0, 0, 3 },
@@ -733,6 +769,10 @@
 	{ 74, HELITAKEOFF, NOTHING_block, 0 },
 	{ 75, HELITAKEOFF, NOTHING_block, 0 },
 	{ 76, 255, RUNWAY_IN_block, 0 }, { 76, 0, RUNWAY_IN_block, 37 },
+	{ 77, 0, NOTHING_block, 46},
+	{ 78, 0, NOTHING_block, 46},
+	{ 79, 0, NOTHING_block, 44},
+	{ 80, 0, NOTHING_block, 44},
 	{ MAX_ELEMENTS, 0, 0, 0 } // end marker. DO NOT REMOVE
 };
 
